### DISCUSSION TITLE
.NET部署后文件夹结构请教

### DISCUSSION THREAD

**Discussion by liesauer** [YerongAI/Office-Tool#526] [Q&A] [answered]
作者你好，首先感谢制作了`Office Tool Plus`这个这么棒的工具，我自己也是.NET开发者，我发现`Office Tool Plus`的目录结构非常简洁，不像正常会携带一堆的dll文件，能请教下你是怎么部署的吗？
![image](https://user-images.githubusercontent.com/8676741/174932656-41f30bd9-44a2-46f5-abe1-7887cf6f674a.png)

**Comment by YerongAI** [comment, marked as answer]
步骤：

1. 将程序发布为单个文件，不包含 runtime.
2. 将 Runtime 从对应的位置进行拷贝，32 位为 `C:\Program Files (x86)\dotnet`，64 位为 `C:\Program Files\dotnet`
3. 具体需要复制的文件参照 Office Tool Plus 的文件夹结构，看看就懂了。

**Reply by liesauer** [reply]
懂了，还想请教下，就是你是怎么知道`主程序 + hostfxr + shared文件夹`这种方式是能运行起来的，自己研究的吗还是参考文章的？因为这种方式官方文档里应该是没有提及的。

**Reply by YerongAI** [reply]
> 懂了，还想请教下，就是你是怎么知道`主程序 + hostfxr + shared文件夹`这种方式是能运行起来的，自己研究的吗还是参考文章的？因为这种方式官方文档里应该是没有提及的。

Link: https://www.virustotal.com/gui/file/81f7bfbb69bcadc6b548ae7bf9297972db868489a14c6acf44166153362d7c8e/behavior/VirusTotal%20Jujubox
See: Files Opened

**Reply by liesauer** [reply]
👍

**Reply by liesauer** [reply]
你好，我想再问一下，这是用了第三方的打包工具吗？这个和官方的`PublishSingleFile`结构完全不一样，能告诉下是通过什么命令生成的吗？

**Reply by YerongAI** [reply]
> 你好，我想再问一下，这是用了第三方的打包工具吗？这个和官方的`PublishSingleFile`结构完全不一样，能告诉下是通过什么命令生成的吗？

这就是官方的。
